### PR TITLE
Fix overzealous toBasetype'ing for debuginfo types

### LINKED ISSUE #3688: Enums are not showing up correctly as members of a struct when debugging.
```
enum B : byte {x,y}

struct Y {B A;}
int main()
{
Y y = { A:B.x };
return 0;
}

```
Set breakpoint at ```return 0;``` and you will see A as A = 0

### PR DESCRIPTION
**Pull request by kinke**
Resolves #3688.

### CODE REVIEW

File: tests/debuginfo/args_cdb.d [hunk 1]
```diff
@@ -134,7 +135,7 @@ int byValue(ubyte ub, ushort us, uint ui, ulong ul,
  */
 size_t byValueShort(Large large)
 {
-// CDB: bp1 /1 `args_cdb.d:137`
+// CDB: bp1 /1 `args_cdb.d:138`
```
Comment (JohanEngelen) [inline tests/debuginfo/args_cdb.d:138]: fyi, FileCheck also gives you `[[@LINE]]` or `[[@LINE+<offset>]]` to check for linenumber output. https://llvm.org/docs/CommandGuide/FileCheck.html#filecheck-pseudo-numeric-variables

Comment (kinke) [reply tests/debuginfo/args_cdb.d:138]: I know, but this isn't run through FileCheck.

Comment (JohanEngelen) [reply tests/debuginfo/args_cdb.d:138]: ah yes of course, sorry for silly remark!

Comment (kinke) [reply tests/debuginfo/args_cdb.d:138]: Don't be, thanks for taking a look!

**Pull request merged**